cli/src/commands/create/comments.rs

```diff
@@ -403,38 +405,40 @@ fn upload_comments_from_reader(
         }
 
         if (comments_to_put.len() + comments_to_sync.len()) >= batch_size {
-            upload_batch(
+            upload_batch_of_comments(
                 client,
                 source,
-                dataset_name,
                 statistics,
                 &comments_to_put,
                 &comments_to_sync,
-                &annotations,
                 &audio_paths,
                 no_charge,
             )?;
             comments_to_put.clear();
             comments_to_sync.clear();
-            annotations.clear();
             audio_paths.clear();
         }
     }
 
     if !comments_to_put.is_empty() || !comments_to_sync.is_empty() {
-        upload_batch(
+        upload_batch_of_comments(
             client,
             source,
-            dataset_name,
             statistics,
             &comments_to_put,
             &comments_to_sync,
-            &annotations,
             &audio_paths,
             no_charge,
         )?;
     }
 
+    if let Some(dataset_name) = dataset_name {
```

Comment:
Doing this after all comments are uploaded to avoid the scenario where each batch only has 3 or 4 annotations in, meaning you'll never be using all cores to upload. Doing it after always means we'll use the max available threads to upload.